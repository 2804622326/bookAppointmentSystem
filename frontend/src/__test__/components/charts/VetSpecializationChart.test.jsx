import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import VetSpecializationChart from '../../../components/charts/VetSpecializationChart.jsx';

jest.mock('recharts', () => ({
  ResponsiveContainer: ({ children }) => <div>{children}</div>,
  BarChart: ({ children }) => <div data-testid='bar-chart'>{children}</div>,
  XAxis: () => <div />,
  YAxis: () => <div />,
  Tooltip: () => <div />,
  Legend: () => <div />,
  Bar: ({ children }) => <div>{children}</div>,
  Cell: () => <div />,
}));

jest.mock('../../../components/utils/utilities', () => ({
  generateColor: jest.fn(() => '#111'),
}));

jest.mock('../../../components/user/UserService.js', () => ({
  aggregateVetBySpecialization: jest.fn(() => Promise.resolve([{ specialization: 'Surgery', count: 2 }])),
}));

jest.mock('../../../components/common/NoDataAvailable.jsx', () => () => <div data-testid='no-data' />);

test('renders vet specialization chart with data', async () => {
  const { asFragment } = render(<VetSpecializationChart />);
  await waitFor(() => screen.getByText(/veterinarians by specializations/i));
  expect(screen.queryByTestId('no-data')).not.toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
