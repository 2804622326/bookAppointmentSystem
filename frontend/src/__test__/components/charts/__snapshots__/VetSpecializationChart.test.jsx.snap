// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders vet specialization chart with data 1`] = `
<DocumentFragment>
  <section>
    <div>
      <h5
        class="mb-4 mt-4 chart-title"
      >
        Veterinarians by Specializations
      </h5>
      <div
        data-testid="bar-chart"
      >
        <div />
        <div />
        <div />
        <div />
        <div>
          <div />
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
